legacy/root/fix_macos_sdk_for_rootcling.patch

```diff
@@ -0,0 +1,119 @@
++   
+    if not existing_ldlib: existing_ldlib = ""
+    my_env["LD_LIBRARY_PATH"] = os.path.join(rootdir, "lib") + ":" + existing_ldlib
++   if osx_sdk:
++      my_env["SDKROOT"] = 'osx_sdk'
+ 
+    ret = subprocess.call(command.split(), env=my_env)
+    if ret == 0:
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 77c7762225..924942293e 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -524,23 +524,44 @@ else()
+   get_property(__pch_dependencies GLOBAL PROPERTY ROOT_PCH_DEPENDENCIES)
+   get_property(__pch_dictionaries GLOBAL PROPERTY ROOT_PCH_DICTIONARIES)
+ 
+-  add_custom_command(OUTPUT etc/allDict.cxx.pch
+-    BYPRODUCTS
+-      etc/dictpch/allCppflags.txt
+-      etc/dictpch/allHeaders.h
+-      etc/dictpch/allLinkDefs.h
+-    COMMAND
+-      ${PYTHON_EXECUTABLE} ${CMAKE_SOURCE_DIR}/build/unix/makepchinput.py
+-      ${CMAKE_SOURCE_DIR} . ${pyroot_legacy} ${__cling_pch}
+-    COMMAND
+-      ${CMAKE_COMMAND} -E env ROOTIGNOREPREFIX=1 ${PYTHON_EXECUTABLE}
+-      ${CMAKE_SOURCE_DIR}/etc/dictpch/makepch.py etc/allDict.cxx.pch
+-      ${__allIncludes} -I${CMAKE_BINARY_DIR}/include -I${CMAKE_SOURCE_DIR}/core
+-    DEPENDS
+-      rootcling ${__pch_dependencies} ${__pch_dictionaries}
+-      ${CMAKE_SOURCE_DIR}/build/unix/makepchinput.py
+-      ${CMAKE_SOURCE_DIR}/etc/dictpch/makepch.py
+-  )
++  if(APPLE AND CMAKE_OSX_SYSROOT)
++    add_custom_command(OUTPUT etc/allDict.cxx.pch
++      BYPRODUCTS
++        etc/dictpch/allCppflags.txt
++        etc/dictpch/allHeaders.h
++        etc/dictpch/allLinkDefs.h
++      COMMAND
++        ${PYTHON_EXECUTABLE} ${CMAKE_SOURCE_DIR}/build/unix/makepchinput.py
++        ${CMAKE_SOURCE_DIR} . ${pyroot_legacy} ${__cling_pch}
++      COMMAND
++        ${CMAKE_COMMAND} -E env "SDKROOT=${CMAKE_OSX_SYSROOT}" env ROOTIGNOREPREFIX=1 ${PYTHON_EXECUTABLE}
++        ${CMAKE_SOURCE_DIR}/etc/dictpch/makepch.py etc/allDict.cxx.pch
++        ${__allIncludes} -I${CMAKE_BINARY_DIR}/include -I${CMAKE_SOURCE_DIR}/core
++      DEPENDS
++        rootcling ${__pch_dependencies} ${__pch_dictionaries}
++        ${CMAKE_SOURCE_DIR}/build/unix/makepchinput.py
++        ${CMAKE_SOURCE_DIR}/etc/dictpch/makepch.py
++    )
++  else()
++    add_custom_command(OUTPUT etc/allDict.cxx.pch
++      BYPRODUCTS
++        etc/dictpch/allCppflags.txt
++        etc/dictpch/allHeaders.h
++        etc/dictpch/allLinkDefs.h
++      COMMAND
++        ${PYTHON_EXECUTABLE} ${CMAKE_SOURCE_DIR}/build/unix/makepchinput.py
++        ${CMAKE_SOURCE_DIR} . ${pyroot_legacy} ${__cling_pch}
++      COMMAND
++        ${CMAKE_COMMAND} -E env ROOTIGNOREPREFIX=1 ${PYTHON_EXECUTABLE}
++        ${CMAKE_SOURCE_DIR}/etc/dictpch/makepch.py etc/allDict.cxx.pch
++        ${__allIncludes} -I${CMAKE_BINARY_DIR}/include -I${CMAKE_SOURCE_DIR}/core
++      DEPENDS
++        rootcling ${__pch_dependencies} ${__pch_dictionaries}
++        ${CMAKE_SOURCE_DIR}/build/unix/makepchinput.py
++        ${CMAKE_SOURCE_DIR}/etc/dictpch/makepch.py
++    )
++  endif()
```

Comment:
For my taste this is a lot of code duplication, only one line adds a new argument?!

Reply:
Will try it. I think the problem was the double quotes which were not properly handled but maybe your version works.